### PR TITLE
Fadi week2

### PR DESCRIPTION
**Pull request by fadialset**
I have an error in exercise 2 and I don't know it but the tables are created and working very well

### CODE REVIEW

File: Week2/homwork/ex2-relationships.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+const CREATE_RESEARCH_PAPERS_TABLE = `
+    CREATE TABLE IF NOT EXISTS research_papers(
+        paper_id INT auto_increment PRIMARY KEY,
+        paper_title VARCHAR(255),
+        conference VARCHAR(255),
+        publish_date DATE
+    );`;
+
+const CREATE_AUTHORS_RESEARCH_TABLE = `
+    CREATE TABLE authors_research_papers(
```
Comment (Breus) [inline Week2/homwork/ex2-relationships.js:22]: ....IF NOT EXISTS....

**Review by Breus — COMMENTED** [review]
> I have an error in exercise 2 and I don't know it but the tables are created and working very well

The error is in the authors_researches.json (you misspelled this btw). You reference to non-existent author_no's (you insert 14 authors in the database from authors.json and insert author_no 15 from this json file). 

For the rest, well done on the homework!

File: Week2/homwork/ex4-aggregate.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    FROM authors
+    GROUP BY university;
+    `;
+
+const SUM_PAPERS_UNIVERSITY = `
+    SELECT university,COUNT(research_papers.paper_title) AS research_papers
+    FROM research_papers
+    LEFT JOIN authors_research_papers ON research_paper_no = research_papers.paper_id
+    LEFT JOIN authors on authors_research_papers.author_no = authors.author_no
+    GROUP BY university;
```
Comment (Breus) [inline Week2/homwork/ex4-aggregate.js:41]: Same thing here, you don't actually need to know the paper_title to get the number of papers. You can just count the number of authors_research_papers.research_paper_no and select from that table without joining on research_papers.

File: Week2/homwork/ex4-aggregate.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    `;
+
+const H_INDEX_MIN_MAX = `
+    SELECT university, MAX(h_index), MIN(h_index)
+    FROM authors
+    GROUP BY university;
+    `;
```
Comment (Breus) [inline Week2/homwork/ex4-aggregate.js:48]: Correct

File: Week2/homwork/ex4-aggregate.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    FROM research_papers
+    LEFT JOIN authors_research_papers ON research_paper_no = research_papers.paper_id
+    LEFT JOIN authors ON authors_research_papers.author_no = authors.author_no
+    WHERE gender = 'f';
+    `;
+
+const H_INDEX_AVERAGE= `
+    SELECT university,AVG(h_index) AS average
+    FROM authors
+    GROUP BY university;
```
Comment (Breus) [inline Week2/homwork/ex4-aggregate.js:33]: Correct

File: Week2/homwork/ex4-aggregate.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    LEFT JOIN authors ON authors_research_papers.author_no = authors.author_no
+    GROUP BY paper_title;
+    `;
+
+const SUM_FEMALE_PAPERS = `
+    SELECT COUNT(research_papers.paper_title) AS females_papers
+    FROM research_papers
+    LEFT JOIN authors_research_papers ON research_paper_no = research_papers.paper_id
+    LEFT JOIN authors ON authors_research_papers.author_no = authors.author_no
+    WHERE gender = 'f';
```
Comment (Breus) [inline Week2/homwork/ex4-aggregate.js:27]: Correct

**Review by Breus — COMMENTED** [review]
Good job on the homework! I put a comment about exercise 2 and in exercise 4 you need to do two queries slightly different for better performance ;-)

File: Week2/homwork/authors.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+[
```
[the remark this answers is not in the document]
Comment (fadialset) [reply Week2/homwork/authors.json:1]: thank you I tried to make it as realistic as possible

File: Week2/homwork/researchPapers.json [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    {"paper_title": "How to forecast and shape behavioral patterns","conference": "Utrecht Conference","publish_date": "2012-09-16"},
+    {"paper_title": "Factors that influence behavior and character","conference": "Dubai Conference","publish_date": "2007-08-08"},
+    {"paper_title": "Mechanisms of aggravating habits","conference": "washington Conference","publish_date": "2008-05-25"},
+    {"paper_title": "Mechanisms of thinking: it’s all in your head","conference": "Amsterdam Conference","publish_date": "2009-12-22"},
+    {"paper_title": "How behavioral patterns develop","conference": "London Conference","publish_date": "2010-10-11"},
+    {"paper_title": "Differences and similarities in the behavioral patterns of diverse cultures","conference": "Berlin Conference","publish_date": "2014-03-23"},
+    {"paper_title": "Is there a genetic link to optimism that can shape behaviors and attitudes?","conference": "Rotterdam Conference","publish_date": "1993-05-03"}
```
[the remark this answers is not in the document]
Comment (fadialset) [reply Week2/homwork/researchPapers.json:30]: it is auto increment that it why i didn't wrote them

File: Week2/homwork/authorsReaserches.json [hunk 8]
```diff
@@ -0,0 +1,42 @@
+[
+    {"author_no": 1, "research_paper_no": 13},
+    {"author_no": 1, "research_paper_no": 14},
+    {"author_no": 1, "research_paper_no": 15},
+    {"author_no": 2, "research_paper_no": 11},
+    {"author_no": 2, "research_paper_no": 12},
+    {"author_no": 3, "research_paper_no": 13},
+    {"author_no": 3, "research_paper_no": 14},
+    {"author_no": 3, "research_paper_no": 15},
```
[the remark this answers is not in the document]
Comment (fadialset) [reply Week2/homwork/authorsReaserches.json:9]: author_no is auto_increment that is why I didnt wrote it in the authors.json file
and the paper_id is a mistake I will fix it .
thanks for the note

File: Week2/homwork/ex1-kyes.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  password: 'hyfpassword',
+  database: 'fadiweek2',
+  port: 3306
+});
+
+
+const execQuery = util.promisify(connection.query.bind(connection));
+
+async function seedDatabase() {
+  const CREATE_AUTHORS_TABLE = `
```
[the remark this answers is not in the document]
Comment (fadialset) [reply Week2/homwork/ex1-kyes.js:16]: thank you

File: Week2/homwork/ex1-kyes.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'fadiweek2',
+  port: 3306
+});
+
+
+const execQuery = util.promisify(connection.query.bind(connection));
+
+async function seedDatabase() {
```
[the remark this answers is not in the document]
Comment (fadialset) [reply Week2/homwork/ex1-kyes.js:15]: yeah I used the same syntax we learned in the reading materials that's why I used the same name

File: Week2/homwork/ex3-joins.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    console.log("names of all authors and their corresponding mentors:");
+    console.log(await execQuery(SELECT_AUTHORS_MENTORS));
+
+    console.log("all columns of authors and their published paper_title:");
+    console.log(await execQuery(SELECT_AUTHORS_PAPERS));
+
+    connection.end();
+  } catch (error) {
+    console.error(error.message);
+    connection.end();
```
[the remark this answers is not in the document]
Comment (fadialset) [reply Week2/homwork/ex3-joins.js:43]: Ok thanks I thought I need to use two .

File: Week2/homwork/ex2-relationships.js (hunk 1 above)
Comment (fadialset) [reply Week2/homwork/ex2-relationships.js:22]: this is the only one I forgot to use IF NOT EXISTS with . thanks for noting

File: Week2/homwork/ex4-aggregate.js (hunk 2 above)
Comment (fadialset) [reply Week2/homwork/ex4-aggregate.js:41]: Ok I will keep it in my mind for  the future thank you

**Pull request closed without merging**